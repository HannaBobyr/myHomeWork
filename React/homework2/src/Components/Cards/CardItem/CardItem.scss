.card-item {
  background-color: aliceblue;
  border-radius: 10px;
  position: relative;
  img {
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  }
  p {
    margin-left: 10px;
  }
  &--name {
    font-weight: 600;
    font-size: 18px;
    line-height: 18px;
  }
  &--price {
    font-size: 20px;
    line-height: 20px;
  }
  &--article {
    font-size: 14px;
    line-height: 14px;
    color: #000;
  }
  &--color {
    font-size: 16px;
    line-height: 16px;
  }
  &--btn {
    cursor: pointer;
    position: absolute;
    bottom: 10px;
    right: 10px;
    text-transform: uppercase;
    border: none;
    border-radius: 7px;
    padding: 7px;
    font-size: 12px;
  }
  &--elect {
    cursor: pointer;
    position: absolute;
    top: 7px;
    right: 10px;
    color: rgb(234, 255, 0);
    font-size: 30px;
    background-color: transparent;
    border: none;
  }
}
