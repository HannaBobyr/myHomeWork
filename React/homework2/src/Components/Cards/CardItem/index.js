import React, { Component } from "react";
import "./CardItem.scss";
import Button from "../../ModalWindow/Button/Button";
import PropTypes from "prop-types";
class CardItem extends Component {
  state = {
    isElected: false,
  };
  handleElectedBtn = () => {
    this.setState({ isElected: true });
  };
  componentDidMount() {
    const { card, arrayElected } = this.props;
    const { article } = card;
    arrayElected.forEach((element) => {
      if (element === article) {
        this.setState({ isElected: true });
      }
    });
  }
  render() {
    const { card, OpenModal, handleArrayElected } = this.props;
    const { name, price, img, article, color } = card;
    return (
      <li className="card-item">
        <img src={img} alt="Logo" width="250px" height="170px" />
        <p className="card-item--name">{name}</p>
        <p className="card-item--price">price: {price}$</p>
        {this.state.isElected === false && (
          <button
            className="card-item--elect"
            onClick={() => {
              this.handleElectedBtn();
              handleArrayElected(article);
            }}
          >
            &#9734;
          </button>
        )}
        {this.state.isElected === true && (
          <button className="card-item--elect"> &#9733;</button>
        )}
        <p className="card-item--article">article: {article}</p>
        <p className="card-item--color">color: {color}</p>
        <Button
          className="card-item--btn"
          onClick={() => OpenModal(article)}
          text="add to cart"
          backgroundColor="antiquewhite"
        />
      </li>
    );
  }
}
export default CardItem;
CardItem.propTypes = {
  handleCounter: PropTypes.func,
  isOpenModal: PropTypes.func,
  card: PropTypes.shape({
    name: PropTypes.string,
    price: PropTypes.number,
    img: PropTypes.string,
    article: PropTypes.number,
    color: PropTypes.string,
  }),
};
