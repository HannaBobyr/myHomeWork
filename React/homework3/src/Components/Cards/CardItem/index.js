import React, { useState, useEffect } from "react";
import "./CardItem.scss";
import Button from "../../ModalWindow/Button/Button";
import PropTypes from "prop-types";
function CardItem({ card, OpenModal, handleArrayElected, arrayElected }) {
  const { name, price, img, article, color } = card;
  const [isElected, setIsElected] = useState(false);
  useEffect(() => {
    const filter = arrayElected.find((el) => el === article);
    if (filter) {
      setIsElected(true);
    }
  }, [arrayElected, article]);
  const handleElectedBtn = () => {
    setIsElected(true);
    handleArrayElected(article);
  };
  return (
    <li className="card-item">
      <img src={img} alt="Logo" width="250px" height="170px" />
      <p className="card-item--name">{name}</p>
      <p className="card-item--price">price: {price}$</p>
      {isElected === false && (
        <button className="card-item--elect" onClick={handleElectedBtn}>
          &#9734;
        </button>
      )}
      {isElected === true && (
        <button className="card-item--elect"> &#9733;</button>
      )}
      <p className="card-item--article">article: {article}</p>
      <p className="card-item--color">color: {color}</p>
      <Button
        className="card-item--btn"
        onClick={() => OpenModal(article)}
        text="add to cart"
        backgroundColor="antiquewhite"
      />
    </li>
  );
}
export default CardItem;
CardItem.propTypes = {
  handleCounter: PropTypes.func,
  isOpenModal: PropTypes.func,
  card: PropTypes.shape({
    name: PropTypes.string,
    price: PropTypes.number,
    img: PropTypes.string,
    article: PropTypes.number,
    color: PropTypes.string,
  }),
};
