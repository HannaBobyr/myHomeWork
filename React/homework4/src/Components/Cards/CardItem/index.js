import React, { useState, useEffect } from "react";
import "./CardItem.scss";
import Button from "Components/ModalWindow/Button/Button";
import { handleArrayElected, openModalAdded } from "store/actions/handleCards";
import { useSelector, useDispatch } from "react-redux";
import { handleWindow } from "store/actions/ModalWindow";
function CardItem({ card }) {
  const { name, price, img, article, color } = card;
  const { arrayElected } = useSelector((state) => state.handleCards);
  const [isElected, setIsElected] = useState(false);
  const dispatch = useDispatch();
  useEffect(() => {
    const filter = arrayElected.find((el) => el === article);
    if (filter) {
      setIsElected(true);
    }
  }, [arrayElected, article]);
  const handleArrayElectedBtn = () => {
    setIsElected(true);
    const checkId = arrayElected.find((el) => el === article);
    if (!checkId) {
      // arrayElected.push(article);
      const newArrayElected = [...arrayElected, article];
      dispatch(handleArrayElected(newArrayElected));
      localStorage.setItem("arrayElected", JSON.stringify(newArrayElected));
    }
  };
  function OpenModal() {
    dispatch(handleWindow(true));
    dispatch(openModalAdded(article));
  }
  return (
    <li className="card-item">
      <img src={img} alt="Logo" width="250px" height="170px" />
      <p className="card-item--name">{name}</p>
      <p className="card-item--price">price: {price}$</p>
      {isElected === false && (
        <button className="card-item--elect" onClick={handleArrayElectedBtn}>
          &#9734;
        </button>
      )}
      {isElected === true && (
        <button className="card-item--elect"> &#9733;</button>
      )}
      <p className="card-item--article">article: {article}</p>
      <p className="card-item--color">color: {color}</p>
      <Button
        className="card-item--btn"
        onClick={OpenModal}
        text="add to cart"
        backgroundColor="antiquewhite"
      />
    </li>
  );
}
export default CardItem;
