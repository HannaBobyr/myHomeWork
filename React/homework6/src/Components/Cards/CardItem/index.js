import React, { useState, useEffect, useContext } from "react";
import "./CardItem.scss";
import Button from "Components/ModalWindow/Button/Button";
import { handleArrayElected, openModalAdded } from "store/actions/handleCards";
import { useSelector, useDispatch } from "react-redux";
import { handleWindow } from "store/actions/ModalWindow";
import { ViewsContext } from "context/view-context";
import { connect } from "react-redux";
function CardItem({ card, arrayElected }) {
  const { name, price, img, article, color } = card;
  //const { arrayElected } = useSelector((state) => state.handleCards);
  const [isElected, setIsElected] = useState(false);
  const dispatch = useDispatch();
  const { view } = useContext(ViewsContext);
  useEffect(() => {
    const filter = arrayElected.find((el) => el === article);
    if (filter) {
      setIsElected(true);
    }
  }, [arrayElected, article]);
  const handleArrayElectedBtn = () => {
    setIsElected(true);
    const checkId = arrayElected.find((el) => el === article);
    if (!checkId) {
      const newArrayElected = [...arrayElected, article];
      dispatch(handleArrayElected(newArrayElected));
      localStorage.setItem("arrayElected", JSON.stringify(newArrayElected));
    }
  };
  function OpenModal() {
    dispatch(handleWindow(true));
    dispatch(openModalAdded(article));
  }
  return (
    <li style={view.item} className="card-item">
      <img src={img} alt="Logo" style={view.img} width="250px" height="170px" />
      <div>
        <p className="card-item--name">{name}</p>
        <p className="card-item--price">price: {price}$</p>
        {isElected === false && (
          <button
            style={view.elected}
            className="card-item--elect"
            onClick={handleArrayElectedBtn}
          >
            &#9734;
          </button>
        )}
        {isElected === true && (
          <button style={view.elected} className="card-item--elect">
            {" "}
            &#9733;
          </button>
        )}
        <p className="card-item--article">article: {article}</p>
        <p className="card-item--color">color: {color}</p>
        <Button
          className="card-item--btn"
          onClick={OpenModal}
          text="add to cart"
          backgroundColor="antiquewhite"
        />
      </div>
    </li>
  );
}
const mapStateToProps = (state) => ({
  arrayElected: state.handleCards.arrayElected,
});
export default connect(mapStateToProps)(CardItem);
