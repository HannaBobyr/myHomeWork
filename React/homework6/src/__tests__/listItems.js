import renderer from "react-test-renderer";
import CardItem from "Components/Cards/CardItem";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";

const middlewares = [];
const mockStore = configureStore(middlewares);
const card = {
  name: "Bedroom Loft",
  price: 123,
  img: "https://meb-online.ru/upload/resize_cache/iblock/009/530_530_140cd750bba9870f18aada2478b24840a/spalnya_teana_variant_2.png",
  article: 1,
  color: "milky",
};
describe("Test Card Items", () => {
  it("should render", () => {
    const store = mockStore({
      handleCards: { arrayElected: [1, 2, 3] },
    });
    render(
      <Provider store={store}>
        <CardItem card={card} />
      </Provider>
    );
  });
  it("Matches snapshot", () => {
    const store = mockStore({
      handleCards: { arrayElected: [1, 2, 3] },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <CardItem card={card} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
