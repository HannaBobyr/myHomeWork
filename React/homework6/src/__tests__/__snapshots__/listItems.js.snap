// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Items Matches snapshot 1`] = `
<li
  className="card-item"
  style={Object {}}
>
  <img
    alt="Logo"
    height="170px"
    src="https://meb-online.ru/upload/resize_cache/iblock/009/530_530_140cd750bba9870f18aada2478b24840a/spalnya_teana_variant_2.png"
    style={Object {}}
    width="250px"
  />
  <div>
    <p
      className="card-item--name"
    >
      Bedroom Loft
    </p>
    <p
      className="card-item--price"
    >
      price: 
      123
      $
    </p>
    <button
      className="card-item--elect"
      onClick={[Function]}
      style={Object {}}
    >
      â˜†
    </button>
    <p
      className="card-item--article"
    >
      article: 
      1
    </p>
    <p
      className="card-item--color"
    >
      color: 
      milky
    </p>
    <button
      className="card-item--btn"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "antiquewhite",
        }
      }
      type="button"
    >
      add to cart
    </button>
  </div>
</li>
`;
